### DISCUSSION TITLE
PNP Search Results - Accented Characters

### DISCUSSION THREAD

**Discussion by albanks** [microsoft-search/pnp-modern-search#3849] [General]
I am using version PnP Modern Search v4.12.2.0 on my SharePoint Search results page. I am having issues with search results that contain accented characters. It appears that the search criteria can only contain two words if it contains an accented character.

PNP Results
-----------------------------
COMPAGNIE FRANÇAISE = yes
FRANÇAISE = yes
COMPAGNIE FRANÇAISE D’ASSURANCE = no
COMPAGNIE FRANCAISE D’ASSURANCE  = yes ( not accented characters )
FRANÇAISE D’ASSURANCE = yes
FRANÇAISE D’ASSURANCE POUR = no
FRANCAISE D’ASSURANCE POUR = yes ( no accented character )
FRANCAISE D’ASSURANCE POUR LE = no

**Comment by kasperbolarsen** [comment]
Can you test if this is an API issue? I recommend using either SP Editor or the SharePoint search query tool for the testing

**Reply by albanks** [reply]
I have tested via the Sharepoint Search Query Tool .

example API. 
https://----------------.sharepoint.com/sites/NDAPlatform/_api/search/query?querytext='FRANCAISE'&trimduplicates=false&clienttype='ContentSearchRegular'

I have also tried it with trim duplicates true with similar (less ) results 

COMPAGNIE FRANÇAISE = yes 38 results
FRANÇAISE = yes 134
FRANCAISE = yes 134 (no accented characters)
COMPAGNIE FRANÇAISE D’ASSURANCE =  yes 21
COMPAGNIE FRANCAISE D’ASSURANCE  = yes 21 ( not accented characters )
FRANÇAISE D’ASSURANCE = yes 29
FRANCAISE D’ASSURANCE = yes 29 ( no accented characters)
FRANÇAISE D’ASSURANCE POUR = yes 28
FRANCAISE D’ASSURANCE POUR = yes 28 ( no accented characters)
FRANÇAISE D’ASSURANCE POUR LE yes 28
FRANCAISE D’ASSURANCE POUR LE = yes 28 ( no accented characters) 

Thanks

**Reply by albanks** [reply]
Looks like PNP Search is not working for more than 2 words if they contain Accented characters

**Reply by kasperbolarsen** [reply]
Any chance you can upload or give access to a csv file containing some test data that will enable us to look into the issue?
If so, we can create some list items with those accented text strings and test on those.

**Reply by albanks** [reply]
I started putting some test data together and it seems my original assessment maybe incorrect. 
I have managed to successfully search with accented characters, Items 2, 3 and 4 below  work but search query string Item 1 doesn't . 

All contain accented characters but only 1 contains an apostrophe in the name which I think splits the phrase up somehow e.g.

Splitting the Search Query 

D’ASSURANCE works
FRANÇAISE D’ASSURANCE works 
COMPAGNIE FRANÇAISE D’ASSURANCE does not work

D’ASSURANCE POUR works
D’ASSURANCE POUR LE does not work 
POUR LE COMMERCE EXTÉRIEUR works 

I think what I can see is that if the search string contains D’ASSURANCE it can only contain two words or the search does not work.


1 - COMPAGNIE FRANÇAISE D’ASSURANCE POUR LE COMMERCE EXTÉRIEUR
2 - PRCL Comunicação e Relações Públicas Lda
3 - Feedzai, Consultadoria e Inovação Tecnológica, S.A.
4 - AZIDUS BRASIL PESQUISA CIENTÍFICA E DESENVOLVIMENTO LTDA


Thanks

**Reply by kasperbolarsen** [reply]
Can you check if it will work if you escape the apostrophe? I am pretty sure it should work.
If that works I guess we should be able to update the code to escape any apostrophes in the query

**Reply by albanks** [reply]
I tried escaping the apostrophe , 
COMPAGNIE FRANÇAISE D\'ASSURANCE this did not return any results
FRANÇAISE D\'ASSURANCE no results

**Reply by kasperbolarsen** [reply]
Did you escape the apostrophe like this this? https://sharepoint.stackexchange.com/questions/197980/escaping-special-characters-in-sharepoint-search-query

**Comment by albanks** [comment]
I tried what I think you said

COMPAGNIE FRANÇAISE D\\'ASSURANCE , this search string does not work
COMPAGNIE FRANÇAISE D\'ASSURANCE ,  this also does not work

**Reply by kasperbolarsen** [reply]
I have looked into this issue and might be able to come up with a fix. Keep a eye on the release

**Reply by albanks** [reply]
Thanks for your help and quick update, it is appreciated.

**Comment by albanks** [comment]
I don't want to complicate matters but the following behaves simalarly in that it only works for 2 symbols at any time . If i try 3 or more it stops working.

摩芮思科技咨询（上海）有限公司

大连软件园股份有限公司Dalian Software Park Co.LTD

山东龙鑫智能科技有限公司

南京盛航博测贸易有限公司

吴谭婷

孙丁凤

广州裂变传媒有限公司

**Reply by kasperbolarsen** [reply]
Most likely it is an API issue.  Will you be able to test the query in a classic search?
